wremnants/syst_tools.py

```diff
@@ -365,30 +388,56 @@ def add_muon_efficiency_unc_hists(results, df, helper_stat, helper_syst, axes, c
             
         df = df.Define(f"effStatTnP_{key}_tensor", helper, [*muon_columns_stat_step, "nominal_weight"])
         name = Datagroups.histName(base_name, syst=f"effStatTnP_{key}")
-        effStatTnP = df.HistoBoost(name, axes, [*cols, f"effStatTnP_{key}_tensor"], tensor_axes = helper.tensor_axes, storage=hist.storage.Double())
+        if addhelicity:
+            helper_helicity, helper_helicity_axes = make_muon_eff_stat_helpers_helicity(helper)
```

Comment:
Instead of creating here a helper for the helicity case, I think it would be better to create the helper for the helicity **instead** of the regular helper_stat and helper_syst (maybe somewhere before the build_graph in the histmaker). Then, this function doesn't need to be modified.